Bottom: ccb4a73ebea1fcf66d5680b28657c9a3262e2886
Top:    725c512d4101b9d06119f88818d859276709e106
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-29 23:02:11 +0100

hid/gtk: Remove the unused function utf8_dup_string


---

diff --git a/src/hid/gtk/gui-utils.c b/src/hid/gtk/gui-utils.c
index 7136fbb..0fcb5d2 100644
--- a/src/hid/gtk/gui-utils.c
+++ b/src/hid/gtk/gui-utils.c
@@ -838,34 +838,3 @@ ghid_scrolled_text_view (GtkWidget * box,
     *scr = scrolled;
   return view;
 }
-
-
-/* If src is not utf8, *dst is converted to utf8.
- */
-gboolean
-utf8_dup_string (gchar ** dst_utf8, const gchar * src)
-{
-  if (!dst_utf8 || (!*dst_utf8 && !src))
-    return FALSE;
-  if (*dst_utf8)
-    {
-      if (src && !strcmp (*dst_utf8, src))
-	return FALSE;
-      g_free (*dst_utf8);
-    }
-  if (src)
-    {
-      if (g_utf8_validate (src, -1, NULL))
-	*dst_utf8 = g_strdup (src);
-      else
-	{
-	  *dst_utf8 = g_locale_to_utf8 (src, -1, NULL, NULL, NULL);
-	  if (!*dst_utf8)
-	    *dst_utf8 = g_strdup (src);
-	}
-    }
-  else
-    *dst_utf8 = NULL;
-
-  return TRUE;
-}
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 5ba5aeb..ea49bee 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -347,7 +347,6 @@ void make_route_style_buttons (GHidRouteStyleSelector *rss);
 /* gui-utils.c
 */
 gboolean dup_string (gchar ** dst, const gchar * src);
-gboolean utf8_dup_string (gchar ** dst_utf8, const gchar * src);
 void free_glist_and_data (GList ** list_head);
 
 ModifierKeysState ghid_modifier_keys_state (GdkModifierType * state);
